plugin/trino-hive/src/main/java/io/trino/plugin/hive/coercions/CoercionUtils.java

```diff
@@ -162,8 +162,14 @@ public static Type createTypeFromCoercer(TypeManager typeManager, HiveType fromH
         if (fromType == REAL && toType instanceof DecimalType toDecimalType) {
             return Optional.of(createRealToDecimalCoercer(toDecimalType));
         }
-        if (fromType instanceof TimestampType && toType instanceof VarcharType varcharType) {
-            return Optional.of(new LongTimestampToVarcharCoercer(TIMESTAMP_NANOS, varcharType));
+        if (fromType instanceof TimestampType) {
+            if (toType instanceof VarcharType varcharType) {
+                return Optional.of(new LongTimestampToVarcharCoercer(TIMESTAMP_NANOS, varcharType));
+            }
+            if (toType instanceof DateType toDateType) {
+                return Optional.of(new TimestampCoercer.LongTimestampToDateCoercer(TIMESTAMP_NANOS, toDateType));
```

Comment:
This code seems off.  The code is assuming the `fromType` is exactly `TIMESTAMP_NANOS`.  If that is what you want, then you should test for that exactly `fromType == TIMESTAMP_NANOS`

Reply:
Also static import?

Reply:
Hive supports `NANOSECOND` precision but in case of Trino we have `HiveTimestampPrecision` which could set to `MILLISECOND` or `MICROSECOND` or `NANOSECOND`. Since we are mimicing hive's behavior during reads we are trying to read the underlying timestamp column with a `NANOSECOND` precision irrespective of  `HiveTimestampPrecision` configured. So irrespective of `HiveTimestampPrecision` we read the timestamp as `NANO`

Reply:
Please also do note in the description of the ticket

> This is a mere adaptation of https://github.com/trinodb/trino/pull/16869 to match timestamp to date coercion